### PR TITLE
Update jpeg-turbo and protobuf to use cmake in superbuild

### PR DESCRIPTION
**Pull request by DylanMcKinney**
This PR changes protobuf and jpeg-turbo to cmake in the superbuild. Tested with clean build on surface and it worked. Want to test on a couple other systems before merging. 
**Edit 7/31:** Now tested for clean builds on Surface, Pascal, and Catalyst. Everything seems to be working

### CODE REVIEW

**Comment by JaeseungYeom** [comment]
I am looking forward to seeing this PR gets merged. I had to run something on surface urgently. So, I tried this and it built successfully. However, when I tried to run, I got runtime error as below:
```
[surface118:mpi_rank_0][cudaipc_init_dynamic] [src/mpid/ch3/channels/mrail/src/gen2/ibv_cuda_ipc.c:486] cuda failed with 11
```
This is probably due to the recent changes to the surface system, and maybe addressed by a following PR.

**Comment by ndryden** [comment]
The CUDA-aware MPI on Surface almost certainly needs to be rebuilt.

File: superbuild/protobuf/CMakeLists.txt [hunk 1]
```diff
@@ -6,42 +6,39 @@ set(PROTOBUF_URL "https://github.com/google/protobuf.git"
...
 
 # Where to install PROTOBUF
 set(PROTOBUF_CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}"
   CACHE PATH "The installation location of PROTOBUF.")
 
+set(PROTOBUF_CMAKE_BUILD_TYPE "${PROTOBUF_CMAKE_BUILD_TYPE}"
+  CACHE STRING "The build type for PROTOBUF.")
+
 # The number of jobs to use when running 'make' for protobuf
 set(PROTOBUF_MAKE_JOBS "8"
```
Comment (benson31) [inline superbuild/protobuf/CMakeLists.txt]: You can kill this line.

File: superbuild/jpeg-turbo/CMakeLists.txt [hunk 2]
```diff
@@ -12,18 +17,6 @@ set (JPEG-TURBO_CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}"
 set(JPEG-TURBO_MAKE_JOBS "8"
```
Comment (benson31) [inline superbuild/jpeg-turbo/CMakeLists.txt]: This can go away too.

**Pull request merged**